### DISCUSSION TITLE
Customise color schema for each chart on dashboard

### DISCUSSION THREAD

**Discussion by simon-b06** [apache/superset#25623] [Q&A / Help] [answered]
Hi all,
I would like to use different color schemas (which are definde in the superset_configy.py "EXTRA_CATEGORICAL_COLOR_SCHEMES") for each chart on the dashboard.
Is this possible? 
Thanks

**Comment by ballyalley-o** [comment, marked as answer]
You can use the label_colors property in the dashboard JSON metadata.

To access the dashboard JSON metadata, go to **Edit dashboard** and then click on the three dots in the top right corner and select Edit properties. Expand the ADVANCED section and you will see the JSON METADATA field.

In the JSON METADATA field, add a new section called **label_colors**. In this section, you can specify the desired colors for each metric/dimension.

_like:_
```
{
  "label_colors": {
    "metric_1": ["#FF0000", "#00FF00"],
    "metric_2": ["#0000FF", "#FFFF00"]
  }
}
```

This will set the color for metric_1 to be red and green, and the color for metric_2 to be blue and yellow.

You can also use the label_colors property to specify different color schemes for different charts. 

_you can add a new section to the label_colors property for each chart._

```
{
  "label_colors": {
    "chart_1": {
      "metric_1": ["#FF0000", "#00FF00"],
      "metric_2": ["#0000FF", "#FFFF00"]
    },
    "chart_2": {
      "metric_1": ["#00FF00", "#FF0000"],
      "metric_2": ["#FFFF00", "#0000FF"]
    }
  }
}
```

This will set the color for metric_1 in chart_1 to be red and green, and the color for metric_2 in chart_1 to be blue and yellow. The colors for metric_1 and metric_2 in chart_2 will be reversed.

Once you have added the label_colors property to the dashboard JSON metadata, save the dashboard. Superset will then use the specified colors for the labels in each chart.

If you are using a custom color scheme, you need to make sure that the color scheme is defined in the superset_config.py file,
by adding the color scheme to the EXTRA_CATEGORICAL_COLOR_SCHEMES list.

 to add a new color scheme called my_color_scheme, you would add the following line to the superset_config.py file:
 
 ```EXTRA_CATEGORICAL_COLOR_SCHEMES['my_color_scheme'] = ['#FF0000', '#00FF00', '#0000FF']```
 
 
 Once you have added the color scheme to the **superset_config.py** file, you can use it in the dashboard JSON metadata.
 
 _To use the my_color_scheme color scheme for the chart_1 chart, you would add the following section to the dashboard JSON metadata:_
 
 ```
{
  "label_colors": {
    "chart_1": {
      "metric_1": "my_color_scheme",
      "metric_2": "my_color_scheme"
    }
  }
}
```

Let me know if this works.

**Reply by simon-b06** [reply]
Hi, Thanks for your very helpful answer.
I still have one issue left. I've made the JSON Metadata Configuration like this:
```
{
  "chart_configuration": {
    "21": {
      "id": 21,
      "crossFilters": {"scope": "global", "chartsInScope": [22]}
    },
    "22": {"id": 22, "crossFilters": {"scope": "global", "chartsInScope": [21]}}
  },
  "global_chart_configuration": {
    "scope": {"rootPath": ["ROOT_ID"], "excluded": []},
    "chartsInScope": [21, 22]
  },
  "color_scheme": "",
  "refresh_frequency": 0,
  "expanded_slices": {},
  "label_colors": {
    "21": {
      "Adult": "#FBD983"
    }
  },
  "timed_refresh_immune_slices": [],
  "cross_filters_enabled": true,
  "default_filters": "{}",
  "filter_scopes": {}
}
```
In my idea the chart with the id "21" should now show the lable "Adult" in yellow, but it's still the old color from de defined color schema.

I've tried different "ids" but no goal.

So we're can I find the correct chart id? It there an other config necessery like the feature flag for the EXTRA_CATEGORICAL_COLOR_SCHEMES ("EXTRA_CATEGORICAL_COLOR_SCHEMES": True)?
Thanks.

**Reply by ballyalley-o** [reply]
you can do that by just simply changing the color_scheme property to custom:

`"custom_scheme":"custom"`

;-) let me know how it went.

**Reply by simon-b06** [reply]
Hi,

Thanks for your proposal.
If I do that the whole Dashboard will have one color scheme.
It still doesen't works for a single charts.

In my case I've two charts. One with the id 21 and one with the id 22.

How would be the configuration for this case?

I appreciate every help. Thanks to all.

**Reply by marianysilva** [reply]
I've tried some approaches **without** success @ballyalley-o 😞 

![Screenshot 2024-10-09 at 11 32 28](https://github.com/user-attachments/assets/82120595-d32a-41b7-9543-fbc2fc109078)

```

 "label_colors": {
    "Day - Total Hours by Day of Week" / "4" / "CHART-cugRx_YEpL": {
      "sum_works_filled_hours": "#E89E67",
      "sum_schedule_filled_hours": "#FFECDE",
      "sum_unfilled_hours": "#D3DADC"
    },
    "Night - Total Hours by Day of Week" / "5" / "CHART-XeYbuDvbPe": {
      "sum_works_filled_hours": "#3374A7",
      "sum_schedule_filled_hours": "#CDE5F6",
      "sum_unfilled_hours": "#D3DADC"
    }
  },
```

![Screenshot 2024-10-09 at 13 51 18](https://github.com/user-attachments/assets/af64ccbc-bb07-47b1-93c9-355ee6d9dddc)

**Reply by marianysilva** [reply]
[Slack discussion](https://apache-superset.slack.com/archives/CCKHMGRRB/p1728490138443129)

![image](https://github.com/user-attachments/assets/a7222c55-f9a1-413a-b31e-5442f73b2b82)

**Comment by Maransatto** [comment]
I'm having trouble trying to set the label_colors to a Mixed chart, it only works on Query A, while Query B is ignored.

**Reply by peanut00** [reply]
Hello,
You should use add (1) to label name for query B, as you can shown:

label_colors": {
    "RNC_217g": "#dc3912",
    "RNC_231d": "#dc3912",
    "RNC_615c": "#dc3912",
    "RNC_214a (1)": "#bcbd22",
    "RNC_229a (1)": "#fdd0a2",
    "RNC_616a (1)": "#fdd0a2"
}
Regards,
